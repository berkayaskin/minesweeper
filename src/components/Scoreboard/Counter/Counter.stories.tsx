import { Meta, Story } from '@storybook/react/types-6-0'
import { Counter, CounterProps } from './Counter'

export default {
  title: 'Scoreboard/Counter',
  component: Counter,
} as Meta

const Template: Story<CounterProps> = (args) => <Counter {...args} />

export const CounterExample = Template.bind({})
CounterExample.args = {
  children: '10',
}
