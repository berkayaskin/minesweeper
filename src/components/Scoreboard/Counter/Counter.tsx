import styled from '@emotion/styled'

export interface CounterProps {
  /**
   * Number in the counter
   */
  children: string
}

export const Counter = ({ children }: CounterProps) => {
  return <Frame>{children}</Frame>
}

const Frame = styled.div`
  display: inline-block;
  padding: 0 0.3vw;
  color: #ec433c;
  border: 0.15vw inset;
  line-height: 2vw;
  letter-spacing: 0.08em;
  background: #333;
  text-shadow: 0 0 0.1vw #ec433c;
`
