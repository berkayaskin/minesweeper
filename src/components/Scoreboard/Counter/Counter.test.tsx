import { render } from '@testing-library/react'
import { Counter } from './Counter'

it('should render counter correctly', () => {
  const { asFragment } = render(<Counter>001</Counter>)
  expect(asFragment()).toMatchSnapshot()
})
