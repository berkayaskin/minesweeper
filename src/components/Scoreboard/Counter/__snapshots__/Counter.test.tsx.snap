// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render counter correctly 1`] = `
<DocumentFragment>
  <div
    class="css-d2q1pa"
  >
    001
  </div>
</DocumentFragment>
`;
